src/git.rs

```diff
@@ -0,0 +1,464 @@
+use crate::meta::git_file_status::GitFileStatus;
+use std::path::{Path, PathBuf};
+
+#[allow(dead_code)]
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash, PartialOrd, Ord, Default)]
+pub enum GitStatus {
+    /// No status info
+    #[default]
+    Default,
+    /// No changes (got from git status)
+    Unmodified,
+    /// Entry is ignored item in workdir
+    Ignored,
+    /// Entry does not exist in old version (now in stage)
+    NewInIndex,
+    /// Entry does not exist in old version (not in stage)
+    NewInWorkdir,
+    /// Type of entry changed between old and new
+    Typechange,
+    /// Entry does not exist in new version
+    Deleted,
+    /// Entry was renamed between old and new
+    Renamed,
+    /// Entry content changed between old and new
+    Modified,
+    /// Entry in the index is conflicted
+    Conflicted,
+}
+
+pub struct GitCache {
+    #[cfg(not(feature = "no-git"))]
+    statuses: Vec<(PathBuf, git2::Status)>,
+}
+
+#[cfg(feature = "no-git")]
+impl GitCache {
+    pub fn new(_: &Path) -> Self {
+        Self {}
+    }
+
+    pub fn get(&self, _filepath: &PathBuf, _is_directory: bool) -> Option<GitFileStatus> {
+        None
+    }
+}
+
+#[cfg(not(feature = "no-git"))]
+impl GitCache {
+    pub fn new(path: &Path) -> GitCache {
+        let _cachedir = std::fs::canonicalize(path).unwrap();
```

Comment:
log error by https://github.com/lsd-rs/lsd/blob/8c62e9ed12ccf7a737bd7fc24ccfb0b36cf7b7da/src/main.rs#L66 and return empty instead of unwrap directly

Reply:
removed line. Only used as log.